src/chart/builder.rs

```diff
@@ -180,7 +201,20 @@ impl<'a, 'b, DB: DrawingBackend> ChartBuilder<'a, 'b, DB> {
             }
         }
 
-        std::mem::swap(&mut drawing_area, splitted[4].as_mut().unwrap());
+        for (id, (_, size)) in self.label_area_inset.iter().zip(self.label_area_size.iter()).enumerate().filter(|(_,(inset, size))| **inset && **size != 0) {
+            let area = splitted[4].as_ref().unwrap();
+            let (w, h) = area.dim_in_pixel();
+            let mut new_area = match id {
```

Comment:
What I am planning to do is uses the `drawing::area::shrink` function if the sizes are negative to construct the label area. 
I am not sure if we need to define this function, but it seems the `shrink` function also works for this purpose.

We can first clone the parent drawing area and then shrink it.

Reply:
Alternatively, we can still use the split, but we can actually discard the one in the middle and use `shrink` to create the plotting area.

Reply:
> 
> 
> What I am planning to do is uses the `drawing::area::shrink` function if the sizes are negative to construct the label area.
> I am not sure if we need to define this function, but it seems the `shrink` function also works for this purpose.
> 
> We can first clone the parent drawing area and then shrink it.
Feel free to copy, change any part. Its important for me that it do the trick.
> 
> 
> Alternatively, we can still use the split, but we can actually discard the one in the middle and use `shrink` to create the plotting area.
As i mentoined earlier negative size is useful itself without labels being inset.